pkg/env/env.go

```diff
@@ -79,6 +80,7 @@ type Interface interface {
 	Domain() string
 	FeatureIsSet(Feature) bool
 	FPAuthorizer(string, string) (refreshable.Authorizer, error)
+	FPNewClientCertificateCredential(string) (*azidentity.ClientCertificateCredential, error)
```

Comment:
we should use `azcore.TokenCredential` interface

Reply:
I will take a look - the current FPauthorizer uses cert based authentication to AAD, so I thought that made the most sense at the time. After our conversation yesterday, it sounds like we want to use the FP certificates to create an azcore.TokenCredential and then authenticate to AppLens using that credential. Is that correct?

Reply:
correct. that being said, `azidentity.ClientCertificateCredential` is also `azcore.TokenCredential`

Reply:
I attempted to use the `azcore.TokenCredential` interface, but ran into an issue with code gen. The mock env file that gets generated imports `github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared` instead of just `github.com/Azure/azure-sdk-for-go/sdk/azcore`. Similarly, all references to `azcore.TokenCredential` are replaced with `shared.TokenCredential`. This causes go validation to fail, unit tests to fail, etc.

A bit of research turned up this [azure-sdk-for-go issue](https://github.com/Azure/azure-sdk-for-go/issues/17450#issuecomment-1095280305). Apparently this is a bug/issue with how older versions of the SDK 
 export symbols. Some aspects of this issue have been fixed but none of those changes are available in the go 1.17 compatible version of the SDK that I am using.

Sticking with `azidentity.ClientCertificateCredential` for now.